ebpf/c/tracer-ebpf.c

```diff
@@ -391,17 +405,29 @@ static int read_conn_tuple(conn_tuple_t* tuple, tracer_status_t* status, struct
     if (family == CONN_V4) {
         bpf_probe_read(&saddr_l, sizeof(u32), ((char*)skp) + status->offset_saddr);
         bpf_probe_read(&daddr_l, sizeof(u32), ((char*)skp) + status->offset_daddr);
+
+        if (!saddr_l || !daddr_l) {
+            return 0;
+        }
     } else {
         bpf_probe_read(&saddr_h, sizeof(saddr_h), ((char*)skp) + status->offset_daddr_ipv6 + 2 * sizeof(u64));
         bpf_probe_read(&saddr_l, sizeof(saddr_l), ((char*)skp) + status->offset_daddr_ipv6 + 3 * sizeof(u64));
         bpf_probe_read(&daddr_h, sizeof(daddr_h), ((char*)skp) + status->offset_daddr_ipv6);
         bpf_probe_read(&daddr_l, sizeof(daddr_l), ((char*)skp) + status->offset_daddr_ipv6 + sizeof(u64));
+
+        if (!(saddr_h || saddr_l) || !(daddr_h || daddr_l)) {
```

Comment:
Should this be `(saddr_h && sadddr_l)`? I think this breaks the `is_ipv4_mapped_ipv6` logic, which can have 0 for the high bits.

Reply:
Aside: Is it possible that 0 for source and dest addresses are valid? For local traffic perhaps? https://en.wikipedia.org/wiki/0.0.0.0

Reply:
> Should this be (saddr_h && sadddr_l)? I think this breaks the is_ipv4_mapped_ipv6 logic, which can have 0 for the high bits.

If the high bits are zero it will evaluate to `!(1 || 0) -> !1 -> false`.  Both values would have to be zero for the statement to evaluate to `!(0 || 0) -> !0 -> true` which allow the early return

Reply:
> Is it possible that 0 for source and dest addresses are valid? For local traffic perhaps?

I haven't noticed any valid traffic excluded by this.  This pattern seems to be repeated in other implementations although we may be inadvertently cargo culting something.